### PR TITLE
Tidy up Webpack options, and allow for creation of source maps.

### PR DESCRIPTION
**Pull request by reason-bv**
Here I'm fixing up the generator code to more consistently accept Webpack options overrides via its configuration, and then enable source map creation. See the added docs below:

---

By passing in Webpack configuration options to the generator, you can override any of the default Webpack configuration that is constructed when generating a scout file. Note that this can absolutely break things, for example if you provide your own plugins array.

A more common use case is to request a source map to be constructed:

```
var scout = require('scoutfile');
scout.generate({
  appModules: [
    {
      name: 'MyApp',
      path: './app/scripts/scout/main.js'
    }
  ],

  // Specify `pretty` to get un-uglified output.
  pretty: true,

  // Specify overridden Webpack options.
  webpackOptions: {
    // Create a source map file main.js.map in addition to the scout file.
    devtool: 'source-map'
  }
}).then(function (sources) {
  // When requesting a source map file in addition to a scout file, an array
  // is returned.
  console.log('Scout: ' + sources[0]);
  console.log('Source map: ' + sources[1]);
});
```

If using the grunt task, a `sourceMapDest` configuration property must be provided in addition to `dest` if the source map is generated as a separate file:

```
grunt.initConfig({
  // ...
  scoutfile: {
    server : {
      src : [
        {
          name : 'MyApp',
          path : './app/scripts/scout/main.js'
        }
      ],
      dest : '.tmp/scripts/scout.js',
      sourceMapDest : '.tmp/scripts/scout.js.map'

      // Override the default `APP` namespace
      namespace: 'APPLICATION',

      // Specify `pretty` to get un-uglified code
      pretty : true,

      // Specify overridden Webpack options.
      webpackOptions: {
        // Create a source map file that will be written to sourceMapDest.
        devtool: 'source-map'
      }
    }
  },
  // ...
});
```

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -150,14 +154,46 @@ module.exports.generate = function generateScout(options, callback) {
...
-          resolve(src.toString());
+          // Send back the generated source (and source map if present).
+          //
+          // For backwards compatibility we only send an array if there is in
+          // fact a source map. Otherwise just return the source as a string.
+          //
+          // The pre-source-map-capable code returned only the source string,
+          // but Promises only allow one return argument so we can't just return
+          // two strings without using a structure like an array. This sort of
+          // situation is exactly why people should just use callbacks.
```
Comment (lawnsea) [inline lib/index.js]: Strike this last sentence, please. As much as I enjoy your dry with, this isn't your blog :)

Comment (reason-bv) [inline lib/index.js]: See, I knew this was a great way to get you to continue to work here without remuneration.

File: lib/index.js [hunk 2]
```diff
@@ -138,7 +143,6 @@ module.exports.generate = function generateScout(options, callback) {
       BluebirdPromise.promisify(compiler.run, compiler)().
         then(function (stats) {
           var error;
-
```
Comment (lawnsea) [inline lib/index.js]: The style in this codebase - with the exception of code we pulled in other from other sources - is to have a blank line after variable declarations.

We should probably enforce this with ESLint.

**Comment by lawnsea** [comment]
:+1: other than the above nits. Superb and thorough work as usual.

**Comment by rmurphey** [comment]
It bears asking whether everything should be controlled directly via `webpackOptions`, given that current shortcuts could end up conflicting with provided `webpackOptions`.  Up until this PR, we've supported specific webpack options by adding a new property and converting it internally to a webpack option. I recognize that approach may rapidly be moving from annoying to absurd, but straying from it does present questions of how to resolve conflicts that aren't dealt with here beyond the cautionary comment.

All of that is to say: I'm OK with landing this, but let's think about how this should work long-term.

File: lib/index.js (hunk 2 above)
Comment (reason-bv) [inline lib/index.js]: Looks like there's a rule for that: http://eslint.org/docs/rules/newline-after-var

We can put that in another PR.

**Comment by reason-bv** [comment]
Ok, items noted corrected.

The `webpackOptions` conflict between what has to be done internally and the desire to specify overrides is an interesting one. It is possible that the 80/20 point is to override all except plugins, and merge that, but really it is hard to say without seeing how people are actually going to use it.
